### PR TITLE
[Docs] Fix field cap descriptions

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/schemas/_core.field_caps.yaml [hunk 1]
```diff
@@ -35,14 +35,14 @@ components:
           $ref: '_common.mapping.yaml#/components/schemas/TimeSeriesMetricType'
         non_dimension_indices:
           description: |-
-            If this list is present in response then some indices have the
-            field marked as a dimension and other indices, the ones in this list, do not.
+            If this list is present in the response, then indexes not contained in this list have the
+            field marked as a dimension. Any indexes in this list are not marked.
```
Comment (Naarcha-AWS) [inline spec/schemas/_core.field_caps.yaml:39]:
```suggestion
            field marked as a dimension. Any indexes contained in this list are not marked as a dimension.
```

File: spec/schemas/_core.field_caps.yaml [hunk 2]
```diff
@@ -35,14 +35,14 @@ components:
           $ref: '_common.mapping.yaml#/components/schemas/TimeSeriesMetricType'
         non_dimension_indices:
           description: |-
-            If this list is present in response then some indices have the
-            field marked as a dimension and other indices, the ones in this list, do not.
+            If this list is present in the response, then indexes not contained in this list have the
```
Comment (natebower) [inline spec/schemas/_core.field_caps.yaml:38]:
```suggestion
            If this list is present in the response, then indexes not contained in the list have the
```

File: spec/schemas/_core.field_caps.yaml [hunk 3]
```diff
@@ -35,14 +35,14 @@ components:
...
+            field marked as a dimension. Any indexes contained in this list are not marked as a dimension.
```
Comment (natebower) [inline spec/schemas/_core.field_caps.yaml:39]:
```suggestion
            field marked as a dimension. Any indexes contained in the list are not marked as a dimension.
```

File: spec/schemas/_core.field_caps.yaml [hunk 4]
```diff
@@ -35,14 +35,14 @@ components:
...
           type: array
           items:
             $ref: '_common.yaml#/components/schemas/IndexName'
         metric_conflicts_indices:
           description: |-
-            The list of indices where this field is present if these indices
+            The list of indexes where this field is present if these indexes
```
Comment (natebower) [inline spec/schemas/_core.field_caps.yaml:45]:
```suggestion
            The list of indexes in which this field is present if the indexes
```

File: spec/schemas/_core.field_caps.yaml [hunk 5]
```diff
@@ -35,14 +35,14 @@ components:
...
             don't have the same `time_series_metric` value for this field.
```
Comment (natebower) [inline spec/schemas/_core.field_caps.yaml:46]:
```suggestion
            don't have the same `time_series_metric` value for the field.
```

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS A few changes. Thanks!

**Comment by Naarcha-AWS** [comment]
Ready for maintainer review.

**Review by dblock — APPROVED** [review]

**Pull request merged**